### ISSUE TITLE
Potential bug when describing structs using their crate path

### ISSUE THREAD

**Issue by magnusja** [juhaku/utoipa#862]
Consider this
```
use external_api::types;

...


    #[derive(OpenApi)]
    #[openapi(paths(routes::rfq::create_rfq_v1), components(schemas(types::rfq::v1::RfQCreation, types::rfq::v1::RfQResponse)))]
    struct ApiDoc;
```

Results in reference errors in the generated json:

```
"paths": {
    "/rfqs": {
      "post": {
        "tags": [
          "routes::rfq"
        ],
        "operationId": "create_rfq_v1",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/types.rfq.v1.RfQCreation" // THIS cant be found
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "Pet found successfully",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/types.rfq.v1.RfQResponse"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "RfQCreation": {  // BECAUSE here it is named differently
        "type": "object",
        "required": [
          "customer",
          "assembly",
          "sourcing",
          "rfq"
        ],
```

**Comment by magnusja** [comment]
Found it in the docs

![image](https://github.com/juhaku/utoipa/assets/1659457/b1d1d5be-2a37-425b-a601-cd369def34df)

**Comment by magnusja** [comment]
```
#[derive(Serialize, Deserialize, Debug, ToSchema)]
#[schema(as = types::rfq::v1::RfQCreation)]
pub struct RfQCreation {
    #[schema(as = types::rfq::v1::CustomerInfo)]
    pub customer: CustomerInfo,
    #[schema(as = types::rfq::v1::AssemblyInfo)]
    pub assembly: AssemblyInfo,
    #[schema(as = types::rfq::v1::SourcingInfo)]
    pub sourcing: SourcingInfo,
    #[schema(as = types::rfq::v1::RfqInfo)]
    pub rfq: RfqInfo,
}
```
This does not work :/

**Comment by magnusja** [comment]
```

**Comment by jegork** [comment]
Also an issue for me

**Comment by juhaku** [comment]
@magnusja @jegork The previous inconsistency with the full path and `as` attribute has been resolved in #1034, This is duplicate for #993
